src/python/pants/base/hash_utils.py

```diff
@@ -49,12 +89,50 @@ def stable_json_hash(obj, digest=None, encoder=None):
   :returns: A stable hash of the given `obj`.
   :rtype: str
 
+  :API: public
+  """
+  return json_hash(obj, digest=digest, encoder=encoder)
+
+
+def _json_encode_for_hash(obj, encoder=None):
+  """Call json.dumps with the settings required for json_hash.
+
+  This is extracted so we can more easily debug hashing issues, which are likely to be in the json
+  encoding logic, not the hashing. Its output should not be relied upon by anything at all.
+  """
+  return json.dumps(obj, ensure_ascii=True, allow_nan=False, sort_keys=True, cls=encoder)
```

Comment:
I don't think this function should exist, but for argument sake - if it does, it at least should be chained below instead of copypasta.

Reply:
Removed!